src/test/java/org/openmrs/maven/plugins/AbstractSdkIntegrationTest.java

```diff
@@ -0,0 +1,173 @@
+package org.openmrs.maven.plugins;
+
+import com.google.common.collect.Lists;
+import org.apache.maven.it.VerificationException;
+import org.apache.maven.it.Verifier;
+import org.apache.maven.it.util.ResourceExtractor;
+import org.apache.maven.shared.utils.io.FileUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.junit.runners.BlockJUnit4ClassRunner;
+import org.openmrs.maven.plugins.model.Artifact;
+import org.openmrs.maven.plugins.model.DistroProperties;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+@RunWith(BlockJUnit4ClassRunner.class)
+public abstract class AbstractSdkIntegrationTest {
+
+    /**
+     * contains name of directory in project's target dir, where integration tests are conducted
+     */
+    static final String TEST_DIRECTORY = "/integration-test";
+    static final String MOJO_OPTION_TMPL = "-D%s=%s";
+
+    /**
+     * contains files in test directory which are not created during tests and will not be cleaned up
+     */
+    List<File> testFilesToPersist;
+    /**
+     * maven utility for integration tests
+     */
+    Verifier verifier;
+    /**
+     * test directory, contains mock files and files created during tests
+     */
+    File testDirectory;
+
+    @Before
+    public void setUp() throws Exception{
+        testDirectory = ResourceExtractor.simpleExtractResources(getClass(), TEST_DIRECTORY);
+        verifier = new Verifier(testDirectory.getAbsolutePath());
+
+        testFilesToPersist = new ArrayList<>(Arrays.asList(testDirectory.listFiles()));
+
+        addTaskParam("interactiveMode","false");
+        addTaskParam("openMRSPath",testDirectory.getAbsolutePath());
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        for(File file : testDirectory.listFiles()){
+            if(!testFilesToPersist.contains(file)){
+                FileUtils.deleteDirectory(file);
+            }
+        }
+    }
+
+    public static String setupTestServer() throws Exception{
+        File testDir = ResourceExtractor.simpleExtractResources(AbstractSdkIntegrationTest.class, TEST_DIRECTORY);
+        Verifier verifier = new Verifier(testDir.getAbsolutePath());
+        String serverId = UUID.randomUUID().toString();
+
+        addTaskParam(verifier, "serverId", serverId);
+        addTaskParam(verifier, "distro", "referenceapplication:2.2");
+        addTaskParam(verifier, "openMRSPath", testDir.getAbsolutePath());
+        addMockDbSettings(verifier);
+        verifier.executeGoal("org.openmrs.maven.plugins:openmrs-sdk-maven-plugin:2.1.3-SNAPSHOT:setup");
```

Comment:
Actually if you look at sdk.properties you will find groupId, artifactId and version in there. See also help command on how version is determined in there.